cmd/argocd/commands/app.go

```diff
@@ -1617,3 +1619,92 @@ func NewApplicationPatchCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
 	command.Flags().StringVar(&patch, "patch", "", "Patch")
 	return &command
 }
+
+func NewApplicationPatchResourceCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
+	var patch string
+	var patchType string
+	var resourceName string
+	var namespace string
+	var kind string
+	var group string
+	var all bool
+	command := cobra.Command{
+		Use:   "patch-resource APPNAME",
+		Short: "Patch resource in an application",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) != 1 {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+			appName := args[0]
+
+			conn, appIf := argocdclient.NewClientOrDie(clientOpts).NewApplicationClientOrDie()
+			defer util.Close(conn)
+			ctx := context.Background()
+			resources, err := appIf.ManagedResources(ctx, &application.ResourcesQuery{ApplicationName: &appName})
+			errors.CheckError(err)
+			liveObjs, err := liveObjects(resources.Items)
+			errors.CheckError(err)
+			objectsToPatch := make([]*unstructured.Unstructured, 0)
+			for i := range liveObjs {
+				obj := liveObjs[i]
+				gvk := obj.GroupVersionKind()
+				if group != "" && kind != gvk.Group {
```

Comment:
Sorry another corner case. Imagine this scenario:
```
argocd app patch APPNAME --group "" --kind Service --patch "{}"
```
This will end up patching a knative Service and not the core service. I think we need to change it like:

```
if c.Flag("group").Changed && group != gvk.Group {
```